resolver/src/main/java/io/netty/resolver/RoundRobinInetAddressResolver.java

```diff
@@ -97,4 +97,17 @@ public void operationComplete(Future<List<InetAddress>> future) throws Exception
             }
         });
     }
+
+    /**
+     * @param anyInt any integer value to covert
+     * @param len max limit of range
+     * @return integer in (0, len] range, using formula {@code (anyInt % len + anyInt) % len}
+     */
+    protected static int toIndex(int anyInt, int len) {
+        int index = anyInt % len;
+        if (index < 0) {
+            index = index + len;
```

Comment:
That's exactly what I meant.
I was able to demostratewhy your function works, negative % always return between -(len + 1) and -1 so adding len to it would make from [1..len - 1], index = 0 is introduced by a divisible counter or zero hence filling the gap and making it always continous and harmonic.

And yes, an amortization for a dumb round robin is not worth the effort.
Thanks for helping fix the other function, would had never found the error as it is unlikely to reach `Long.MAX_VALUE` with few years.

Not even my processing 30m increments per second, but at least it is now theoretically correct.